File: app/src/main/java/com/sano/reditto/data/server/APIClient.java
package com.sano.reditto.data.server;

import com.sano.reditto.data.server.pojo.AccessToken;
import com.sano.reditto.data.server.pojo.Top;
import retrofit2.Call;
import retrofit2.http.*;

public interface APIClient {
    @GET("top")
    Call<Top> getTop(
            @Query("t") String time,
            @Query("after") String after,
            @Query("before") String before,
            @Query("count") Integer count,
            @Query("include_categories") Boolean category,
            @Query("limit") Integer limit);
}

File: app/src/main/java/com/sano/reditto/data/server/APIClient.kt
package com.sano.reditto.data.server

import com.sano.reditto.data.server.pojo.Link
import com.sano.reditto.data.server.pojo.Listing
import com.sano.reditto.data.server.pojo.Thing
import io.reactivex.Single
import retrofit2.http.GET
import retrofit2.http.Query

interface APIClient {
    @GET("top")
    fun getTop(
        @Query("t") time: String,
        @Query("after") after: String?,
        @Query("count") count: Int,
        @Query("include_categories") category: Boolean,
        @Query("limit") limit: Int
    ): Single<Thing<Listing<Link>>>
}
